File: javascript/json-transform/src/functions/TransformerFunctionLower.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";

class TransformerFunctionLower extends TransformerFunction {
  constructor() {
    super({});
  }

  override async apply(context: FunctionContext): Promise<any> {
    const result = await context.getString(null);
    return result == null ? null : result.toLowerCase();
  }
}

export default TransformerFunctionLower;

File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionLower.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.FunctionContext;
import co.nlighten.jsontransform.functions.common.TransformerFunction;

public class TransformerFunctionLower extends TransformerFunction {
    public TransformerFunctionLower() {
        super();
    }
    @Override
    public Object apply(FunctionContext context) {
        var result = context.getString(null);
        return result == null ? null : result.toLowerCase();
    }
}
